pythia/seo/tablematcher.py

```diff
@@ -0,0 +1,202 @@
+
+        Returns
+        -------
+        result: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains indices of rows from df_2 that best correspond to rows from df_1.
+        match_score: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains match score for  corresponding best matches.
+        """
+
+        try:
+            from sklearn.metrics.pairwise import cosine_similarity
+        except ImportError:
+            raise SunpyUserWarning("Table Matcher requires Scikit Learn to be installed")
+
+        cosine = cosine_similarity(X=df_1, Y=df_2)
+        result = np.argmax(cosine, axis=1)
+        match_score = np.max(cosine, axis=1)
+
+        return result, match_score
+
+    def match_euclidean(self, df_1, df_2):
+        """
+        Finds euclidean distance between the rows of the two dataframes.
+
+        Parameters
+        ----------
+        df_1: `pd.DataFrame`
+            First DataFrame to match the rows from.
+        df_2: `pd.DataFrame`
+            Second DataFrame to match the rows from.
+
+        Returns
+        -------
+        result: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains indices of rows from df_2 that best correspond to rows from df_1.
+        match_score: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains match score for  corresponding best matches.
+        """
+        try:
+            from sklearn.metrics.pairwise import euclidean_distances
+        except ImportError:
+            raise SunpyUserWarning("Table Matcher requires Scikit Learn to be installed")
+
+        euclidean = euclidean_distances(X=df_1, Y=df_2)
+        result = np.argmin(euclidean, axis=1)
+        match_score = np.min(euclidean, axis=1)
+
+        return result, match_score
+
+    def verify(self, match_score, threshold):
+        """
+        Verify matching quality. If any match score is less than the threshold,
+        raises Sunpy User Warnings.
+
+        Parameters
+        ----------
+        match_score: `numpy.ndarray`
+            Array of size `(n,)` where n is the number of rows in df_1.
+            Contains match score for  corresponding best matches.
+        threshold: `float`
+            Minimum score for considering a proper match.
+        """
+        if self.match_type == 'euclidean':
+            for index, score_value in enumerate(match_score):
+                if score_value > threshold:
+                    warnings.warn(SunpyUserWarning(f"\nMatch at Index {index} is likely to be incorrect\n"))
+
+        if self.match_type == 'cosine':
+            for index, score_value in enumerate(match_score):
+                if score_value < threshold:
+                    warnings.warn(SunpyUserWarning(f"\nMatch at Index {index} is likely to be incorrect\n"))
```

Comment:
All the above is almost equal for both. Think on a way to make it DRY-ier. A dictionary may come handy here ;)